src/mesh/logical_location.hpp

```diff
@@ -251,15 +251,25 @@ struct block_ownership_t {
 inline block_ownership_t
 DetermineOwnership(const LogicalLocation &main_block,
                    const std::set<LogicalLocation> &allowed_neighbors,
-                   const RootGridInfo &rg_info = RootGridInfo()) {
+                   const RootGridInfo &rg_info = RootGridInfo(),
+                   const std::set<LogicalLocation> &newly_refined = {}) {
   block_ownership_t main_owns;
 
-  auto ownership_less_than = [](const LogicalLocation &a, const LogicalLocation &b) {
+  auto ownership_level = [&](const LogicalLocation &a) {
+    // Newly-refined blocks are treated as higher-level than blocks at their
+    // parent level, but lower-level than previously-refined blocks at their
+    // current level.
+    if (newly_refined.count(a)) return 2 * a.level() - 1;
+    return 2 * a.level();
+  };
+
+  auto ownership_less_than = [ownership_level](const LogicalLocation &a,
```

Comment:
still cheeky IMO

Reply:
As in, I should collapse it?  This seemed pretty easy to understand, if a little over-engineered

Reply:
no no---it's fine. Cheeky is not a bad thing.

Reply:
no no---it's fine. Cheeky is not a bad thing.